src/core/reference/include/openvino/reference/reduce_sum.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <numeric>
+
+#include "openvino/core/type/bfloat16.hpp"
+#include "openvino/core/type/float16.hpp"
+#include "openvino/reference/utils/coordinate_transform.hpp"
+#include "openvino/reference/utils/type_util.hpp"
+#include "shape_util.hpp"
+
+namespace ov {
+namespace reference {
+namespace details {
+
+template <typename T, typename std::enable_if<std::is_floating_point<T>::value, bool>::type = true>
+bool isfinite(T x) {
+    return std::isfinite(x);
+}
+
+template <
+    typename T,
+    typename std::enable_if<std::is_same<T, bfloat16>::value || std::is_same<T, float16>::value, bool>::type = true>
+bool isfinite(T x) {
+    return std::isfinite(static_cast<float>(x));
+}
+
+///
+/// \brief      Performs one element summation based on Kahan algorithm to
+/// significantly reduce
+///             the numerical error.
+///
+/// \param[in]  elem            Element to add into the accumulator.
+/// \param      compensation    Variable that accumulates the error.
+/// \param      sum             Result of compensated summation.
+///
+template <class T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+constexpr T kahan_summation(const T in, const T prev_sum, T&) {
+    return in + prev_sum;
+}
+
+template <class T, typename std::enable_if<ov::is_floating_point<T>()>::type* = nullptr>
+T kahan_summation(const T in, const T prev_sum, T& compensation) {
```

Comment:
(detail) Looks like the params description above is not fully aligned with the function.